evalit/rclone/rclone_automation.py

```diff
@@ -107,56 +106,62 @@ def run_automation(self):
-            for line in f:
-                if (
-                    line.find("multipart upload starting chunk 1 size") != -1
-                    or line.find("Transferring unconditionally") != -1
-                ):
-                    if self.debug:
-                        logger.debug(
-                            line.split(":")[3].strip(), line.split("DEBUG")[0].strip()
-                        )
-                    utc_time = datetime.strptime(
-                        line.split("DEBUG")[0].strip(), "%Y/%m/%d %H:%M:%S"
-                    )
-                    start_time_map[line.split(":")[3].strip()] = utc_time
-
-                if line.find("Copied") != -1:
-                    if self.debug:
-                        logger.debug(
-                            line.split(":")[3].strip(), line.split("INFO")[0].strip()
-                        )
-                    utc_time = datetime.strptime(
-                        line.split("INFO")[0].strip(), "%Y/%m/%d %H:%M:%S"
-                    )
-                    end_time_map[line.split(":")[3].strip()] = utc_time
-
-        final_time_array = []
-
-        for key, start_time in start_time_map.items():
-            end_time = end_time_map[key]
-            logger.debug(f"{key}, {(end_time - start_time).total_seconds()}")
-            final_time_array.append(
-                [
-                    (start_time - datetime(1970, 1, 1)).total_seconds(),
-                    (end_time - datetime(1970, 1, 1)).total_seconds(),
-                ]
-            )
-
-        return final_time_array
+        # this deletes the temp file also
+        logger.info(f"Removing temp config at {rclone_config_file.name}")
+        rclone_config_file.close()
+
+        # start_time_map, end_time_map = self.parse_log(rclone_log_file, debug=self.debug)
+        vals = self.parse_log(rclone_log_file, debug=self.debug)
+        logger.debug(
+            f"Delta time for {self.__classname__} = {time.time() - start_automation}"
+        )
+        return vals
+
+    def parse_log(
+        self, log: Union[str, TextIO], debug: bool = False
+    ) -> Tuple[TransferDTO]:
+        # in case it's a path
+        if isinstance(log, str):
+            log = open(log)
+        log.seek(0)
+
+        logger.debug(f"Parsing log at {log.name}")
+
+        timekeeper = {}
+        for line in log:
+            if (
+                line.find("multipart upload starting chunk 1 size") != -1
+                or line.find("Transferring unconditionally") != -1
+            ):
+                fname = line.split(":")[3].strip()
+                dto = timekeeper.get(
+                    fname, TransferDTO(fname=fname, transferer="rclone")
+                )
+                dto.start_time = datetime.strptime(
+                    line.split("DEBUG")[0].strip(), "%Y/%m/%d %H:%M:%S"
+                )
+                timekeeper[fname] = dto
+
+            if line.find("Copied") != -1:
```

Comment:
Should we use a constant for -1? Like `NOT_FOUND_ERROR_CODE = -1`

`.find()` is common enough that people should know this automatically, but the -1 does make a casual reader pause for a second.